### PR TITLE
Create pantheon-yml.md

### PR DESCRIPTION
**Pull request by bmackinney**
@joshkoenig @greg-1-anderson please review

### CODE REVIEW

File: source/docs/articles/sites/code/pantheon-yml.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        script: private/scripts/slack_deploy_notification.php
+```
+## How it Works
+
+1. A developer adds a `pantheon.yml` file to the root of the code repository and attempts to commit it on the server or push a local commit to the platform.
+2. Using a pre-receive git-hook, the platform recognizes its existence and validates it. 
+3. If the file contents are invalid, it will reject the commit.
+4. Once committed, the platform makes a copy of the file and sends it to our automation infrastructure provider. 
+5. When developers run workflows identified in the file’s `workflows:` section, it will perform the user-specified operations before or after the platform workflow runs. 
+6. If project configurations have changed, the platform will deploy those changes to the site environments.
```
Comment (greg-1-anderson) [inline source/docs/articles/sites/code/pantheon-yml.md]: pantheon.yml is per-environment.

"If project configuration has changed, the platform will deploy those changes to the environment where the pantheon.yml file was deployed.

File: source/docs/articles/sites/code/pantheon-yml.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## About `pantheon.yml`
+The file can contain two collections, `api version:` and `workflows:`.  
+
+### API Version
+The current API version for pantheon.yml is `1`. This is the only valid value to place in this field. Additional valid values may be introduced in the future.
+```yaml
+api_version: 1
+```
+### Workflows 
+The workflows collection allows developers to hook into platform operations. Developers Identify the workflows they want to hook into, the workflow “stage” (before or after) to hook into, and then define a sequence of mapping blocks that define the operations the platform will run during the workflow stage. For more information, see [Quicksilver Platform Hooks](/docs/articles/sites/quicksilver/).
```
Comment (greg-1-anderson) [inline source/docs/articles/sites/code/pantheon-yml.md]: Quicksilver platform hooks doc doesn't exist yet?

**Comment by greg-1-anderson** [comment]
Looks good.

File: source/docs/articles/sites/code/pantheon-yml.md (hunk 2 above)
Comment (bmackinney) [inline source/docs/articles/sites/code/pantheon-yml.md]: yeah.. our in-site docs links are relative to the production output state, not the source repository state. We have a symlink in production, for images, that would make these links work in both places, if we were to change all docslinks from `docs/articles...` to `source/docs/articles`. I'm glad you found it.

**Comment by greg-1-anderson** [comment]
:+1:

**Comment by joshkoenig** [comment]
Can we add "example output" that we expect people to see?

**Comment by rachelwhitton** [comment]
Validation output added @bmackinney
@nataliejeremy let me know if you'd rather not include a screenshot for this
+1

**Pull request merged**